swagger-templates/src/main/java/com/okta/swagger/codegen/AbstractOktaJavaClientCodegen.java

```diff
@@ -113,6 +108,74 @@ public void preprocessSwagger(Swagger swagger) {
         buildDiscriminationMap(swagger);
     }
 
+    /**
+     * Figure out which models are top level models (directly returned from a endpoint).
+     * @param swagger The instance of swagger.
+     */
+    protected void buildTopLevelResourceList(Swagger swagger) {
```

Comment:
Method seems too complex for someone reading it for the first time.
Can we add more comments/javadoc for this class/method?